apps/web/src/components/SearchModal/CommonBases.tsx

```diff
@@ -62,42 +67,64 @@ export default function CommonBases({
   selectedCurrency?: Currency | null,
   onSelect: (currency: Currency) => void
 }) {
-  const [ commonCollections, setCommonCollections ] = useState<ERC721Token[]>()
+  const [commonCollections, setCommonCollections] = useState<(ERC721Token | Erc20WrappedNftToken)[]>()
   const native = useNativeCurrency()
   const { t } = useTranslation()
   const pinTokenDescText = commonBasesType === CommonBasesType.SWAP_LIMITORDER ? t('Common tokens') : t('Common bases')
-
-  const {collections } = useSearchCollections(chainId, chainId ? SUGGESTED_BASES_ERC721[chainId] || [] : [])
-  const { data } = useSigner()
+  const provider = useProviderOrSigner(true)
   useMemo(async () => {
-      if (collections?.length && chainId) {
-        const newCollection = collections.map(async (collection) => {
-          const contract = getErc721Contract(collection.id, data )
-          const symbol = await contract.symbol()
-
-          return new ERC721Token(
-            chainId,
-            collection.id,
-            symbol,
-            collection.name,
-            collection.banner,
-            collection.image,
-          )
-        })
-        setCommonCollections(await Promise.all(newCollection))
+    if (!chainId) {
+      return
+    }
+
+    const collections = await ReservoirService().searchCollections(chainId, COMMON_COLLECTIONS[chainId].map((collection) => collection.address))
+
+    const newCollection = collections.map(async (collection) => {
+      const factoryContract = getIUniswapV2FactoryContract(FACTORY_ADDRESS[chainId], provider)
+      const wrapperAddress = await factoryContract.getWrapper(collection.id)
+      const contract = getErc721Contract(collection.id, provider)
+      const symbol = await contract.symbol()
+
+      // Return wrapped token if it exists for the collection
+      if (wrapperAddress !== '0x0000000000000000000000000000000000000000') {
+        const wrapperContract = getErc20WrapperContract(wrapperAddress, provider)
+
+        const decimals = await wrapperContract.decimals()
+
+        return new Erc20WrappedNftToken(
+          chainId,
+          wrapperAddress,
+          decimals,
+          symbol,
+          collection.name,
+          `${collection.image}`,
+          isAddress(collection.id) || collection.id
```

Comment:
Why do you are using  this expression

```suggestion
          isAddress(collection.id) || collection.id
```
? 

Maybe you can use only `isAddress(collection.id)`

Reply:
I had to do this because `isAddress` return a string or boolean in case of failure. The boolean was not accepted as param, then I set the OR for that. But I have thinked of better option